lib/repositories/i_movie_repository.dart

```diff
@@ -0,0 +1,78 @@
+import 'dart:convert';
+
+import 'package:flutter/services.dart';
+
+import '/data_classes/movie.dart';
+
+abstract class MovieRepository {
+  Future<List<Movie>> getMovies();
+
+  Future<Movie> getMovieById(int id);
+}
+
+abstract class GenresRepository {
+  Future<String?> getFilm(int id);
+
+  Future<Map<int, String>> getFilms();
+}
+
+class GenresFromJson extends GenresRepository {
+  static late Map<int, String> genresById;
+  static const genresPath = 'jsonData/genres.json';
+
+  GenresFromJson() {
+    genresById = {};
+  }
+
+  @override
+  Future<String?> getFilm(int id) async {
+    if (genresById.isEmpty) {
+      genresById.addAll(await parseJsonGenres());
+    }
+    return genresById[id];
+  }
+
+  static Future<Map<int, String>> parseJsonGenres() async {
+    var jsonPath = await rootBundle.loadString('jsonData/genres.json');
+    Map<String, dynamic> json = jsonDecode(jsonPath);
+    return {for (var e in json['genres']) e['id'] as int: e['name']};
+  }
+
+  @override
+  Future<Map<int, String>> getFilms() async {
+    if (genresById.isEmpty) {
+      genresById.addAll(await parseJsonGenres());
+    }
+    return genresById;
+  }
+}
+
+class MoviesFromJson extends MovieRepository {
+  static const moviesPath = 'jsonData/movies.json';
+  static late List<Movie> parsedMovies;
+
+  static Future<List<Movie>> parseMovies() async {
```

Comment:
Why `static`?

Reply:
Wouldn´t it be more efficient being the methods static? Anyway, I just need one instance of the repo.
Should I change it to no static?